### ISSUE TITLE
Local NodePort connectivity test has faulty selection dst pod logic 

### ISSUE THREAD

**Issue by christarazi** [cilium/cilium-cli#186]
I am observing a scenario where the local nodeport test selects the wrong dst pod to test connectivity against.

Src pod:  `client2`
Src node:  `10.0.194.76`

```
❯ k -n cilium-test get pods -o custom-columns=NAME:.metadata.name,IP:.status.podIP,NODE:.status.hostIP
NAME                               IP              NODE
client-68c6675687-whsgk            10.128.9.1      10.0.179.183
client2-6c6cdcd976-t48ln           10.128.11.1     10.0.194.76
echo-other-node-588bf78fbb-4q8lz   10.128.11.203   10.0.194.76
echo-same-node-66589c9569-q4vcb    10.128.9.145    10.0.179.183
```

The CLI should select the dst pod as `echo-other-node` as it's on the same node as the src pod.

However, the CLI selects `31646` as the dst port (because it's a nodeport test) which means the dst pod is incorrectly selected as `echo-same-node`.

```
❯ k -n cilium-test get svc -o wide
NAME              TYPE       CLUSTER-IP       EXTERNAL-IP   PORT(S)       
echo-other-node   NodePort   172.30.167.139   <none>        8080:32564/TCP
echo-same-node    NodePort   172.30.35.187    <none>        8080:31646/TCP
```

In summary, the traffic looks like this:

```
client2 -> echo-same-node
10.128.11.1:src-port -> 10.0.194.76:31646
```

where it should actually look like this:

```
client2 -> echo-other-node
10.128.11.1:src-port -> 10.0.194.76:32564
```

This causes the connectivity to fail as a false positive.
